client/finality-grandpa/src/environment.rs

```diff
@@ -400,6 +419,17 @@ impl<B, E, Block: BlockT, N: NetworkT<Block>, RA, SC, VR> Environment<B, E, Bloc
 		self.voter_set_state.with(|voter_set_state| {
 			if let Some(set_state) = f(&voter_set_state)? {
 				*voter_set_state = set_state;
+
+				if let Some(metrics) = self.metrics.as_ref() {
```

Comment:
Using `environment.rs` as a source of truth for the current round was a guess from my side. @andresilva could you approve that that is a good idea?